oshi-core/src/main/java/oshi/util/ConfigUtil.java

```diff
@@ -0,0 +1,164 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package oshi.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The global configuration utility.
+ * 
+ * @author cilki
+ * @since 4.0.0
+ */
+public final class ConfigUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConfigUtil.class);
+
+    private static final Properties configuration = new Properties();
+
+    static {
+        try (InputStream in = ConfigUtil.class.getResourceAsStream("/default.properties")) {
+            if (in != null)
+                configuration.load(in);
+            else
+                LOG.warn("Default configuration not found");
+        } catch (IOException e) {
+            LOG.warn("Failed to load default configuration");
+        }
+    }
+
+    /**
+     * Get the {@code String} property associated with the given key.
+     * 
+     * @param key The property key
+     * @return The property value or {@code null} if not found
+     */
+    public static String get(String key) {
+        return configuration.getProperty(key);
+    }
+
+    /**
+     * Get the {@code String} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the default if not found
+     */
+    public static String get(String key, String def) {
+        return configuration.getProperty(key, def);
+    }
+
+    /**
+     * Get the {@code Integer} property associated with the given key.
+     * 
+     * @param key The property key
+     * @return The property value or {@code 0} if not found
+     */
+    public static int getInteger(String key) {
+        return getInteger(key, 0);
+    }
+
+    /**
+     * Get the {@code Integer} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the default if not found
+     */
+    public static int getInteger(String key, int def) {
+        String value = configuration.getProperty(key);
+        return value == null ? def : Integer.parseInt(value);
+    }
+
+    /**
+     * Get the {@code Double} property associated with the given key.
+     * 
+     * @param key The property key
+     * @return The property value or {@code 0.0} if not found
+     */
+    public static Double getDouble(String key) {
+        return getDouble(key, 0.0);
+    }
+
+    /**
+     * Get the {@code Double} property associated with the given key.
+     * 
+     * @param key The property key
+     * @param def The default value
+     * @return The property value or the default if not found
+     */
+    public static Double getDouble(String key, double def) {
```

Comment:
Since the `String` getter doesn't include the type, why not just use `get()` here as well?